### PR TITLE
Form 21-8940: Ability to upload Supporting Documents#14542

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
Form 21-8940: Ability to upload Supporting Documents

As a Veteran submitting a claim for Individual Unemployability on the Vets.gov platform, I need the ability to upload Supporting Documents, so that I can provide additional evidence associated to my Individual Unemployability while completing the Form 21-8940 application interview.

## Testing done
- [x] local testing
- [x] e2e tests

## Screenshots
screen shot 2018-12-14 at 3 38 53 pm

## Acceptance criteria
- [x] The user is able to answer “Do you have supporting documents you would like to upload?” {radio}
Yes
No
- [x] The user is able to upload supporting documents
- [x] The user is able to delete a file after uploading it
- [x] The user is able to move to the next step in the process, Form 21-8940: Ability to provide Certification #14543
- [x] Field validations (if applicable per the linked document for Validations above) are enforced in the UI
- [x] Error messages (if applicable per the linked document for Validations above) are displayed in the UI

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 1]
```diff
@@ -392,6 +394,24 @@ const formConfig = {
...
+          schema: uploadUnemployabilitySupportingDocumentsChoice.schema,
+        },
+        uploadUnemployabilitySupportingDocuments: {
+          title: 'Upload supporting documents',
+          path: 'upload-unemployability-supporting-documents',
+          depends: formData =>
+            formData['view:unemployabilityUploadChoice'] ===
+              'answerQuestions' &&
+            formData['view:uploadUnemployabilitySupportingDocumentsChoice'],
+          uiSchema: uploadUnemployabilitySupportingDocuments.uiSchema,
```
Comment (imeecuison) [inline src/applications/disability-benefits/all-claims/config/form.js]: Maybe put this depends in utils? Something like this:
```
export const isUploadingSupportingDocuments = formData =>
  _.get('view:unemployabilityUploadChoice', formData, ' ') === 'answerQuestions' &&   _.get('view:uploadUnemployabilitySupportingDocumentsChoice', formData, true)
```

**Review by imeecuison — APPROVED** [review]
Looks good! Just one suggestion about moving depends to utils.

**Comment by zurbergram** [comment]
@department-of-veterans-affairs/code-reviewers-for-external-teams-group - Could I get a code review on this at your earliest convenience? Thanks!

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 2]
```diff
@@ -647,3 +647,7 @@ export const ancillaryFormUploadUi = (label, itemDescription) =>
     },
     attachmentName: false,
   });
+
+export const isUploadingSupporting8940Documents = formData =>
```
Comment (rianfowler) [inline src/applications/disability-benefits/all-claims/utils.jsx]: This may need a unit test.

**Review by rianfowler — APPROVED** [review]
looks good- just one minor suggestion.

**Pull request merged**